pages/api/internal-jobs/email-digest/index.ts

```diff
@@ -0,0 +1,270 @@
+    const subject = userConfig.emailDigest === EmailDigestSettingTypes.DAILY ?
+      `Daily Digest - ${todaysDatePretty}` :
+      `You have ${notificationsCount} new notification${notificationsCount !== 1 ? 's' : ''}`;
+
+    const title = `Welcome to the Pathology daily digest for ${todaysDatePretty}.`;
+    const body = getEmailBody(levelOfDay, notificationsCount, title, user);
+    const sentError = await sendMail(batchId, EmailType.EMAIL_DIGEST, user, subject, body);
+
+    if (!sentError) {
+      sentList.push(user.email);
+    }
+    else {
+      failedList.push(user.email);
+    }
+  }
+
+  return { sentList, failedList };
+}
+
+export async function sendEmailReactivation(batchId: ObjectId, totalEmailedSoFar: string[] = []) {
+  // if they haven't been active in 7 days and they have an email address, send them an email, but only once every 90 days
+  // get users that haven't been active in 7 days
+  const levelOfDay = await getLevelOfDay();
+  const usersThatHaveBeenSentReactivationInPast30d = await EmailLogModel.find({
+    type: EmailType.EMAIL_7D_REACTIVATE,
+    createdAt: { $gte: new Date(Date.now() - 90 * 24 * 60 * 60 * 1000) },
+  }).distinct('userId');
+
+  const inactiveUsers = await UserModel.aggregate([
+    {
+      $match: {
+        _id: { $nin: usersThatHaveBeenSentReactivationInPast30d },
+
+        last_visited_at: { $lte: (Date.now() / 1000) - (7 * 24 * 60 * 60 ) }, // TODO need to refactor last_visited_at to be a DATE object instead of seconds
+
+        email: { $ne: null },
+      },
+    },
+    {
+      $lookup: {
+        from: 'userconfigs',
+        localField: '_id',
+        foreignField: 'userId',
+        as: 'userConfig',
+      },
+    },
+    {
+      $unwind: {
+        path: '$userConfig',
+        preserveNullAndEmptyArrays: false, // if user has no config, don't include them
+      },
+    },
+    {
+      $match: {
+        'userConfig.emailDigest': { $ne: EmailDigestSettingTypes.NONE },
+      },
+    },
+    {
+      $project: {
+        _id: 1,
+        name: 1,
+        email: 1,
+        userConfig: 1,
+        score: 1
+      }
+    }
+  ]);
+
+  const sentList: string[] = [];
+  const failedList: string[] = [];
+  const totalLevels = await LevelModel.countDocuments({
+    isDraft: false,
+  });
+  const totalCreators = await LevelModel.distinct('userId').countDocuments();
+
+  for (const user of inactiveUsers) {
+    const totalLevelsSolved = user.score || 0;
+    const toSolve = (totalLevels - totalLevelsSolved);
+    const subject = 'New Pathology levels are waiting to be solved!';
+    const title = 'We haven\'t seen you in a bit!';
+    const message = `You've completed ${totalLevelsSolved.toLocaleString()} levels on New Pathology. There's ${toSolve.toLocaleString()} levels for you to play by ${totalCreators.toLocaleString()} different creators. Come back and play!`;
+    const body = getEmailBody(levelOfDay, 0, title, user, message);
```

Comment:
0 unread notifications is not necessarily true here

Reply:
True. In this i passed 0 to disable the notifications text from being shown.. maybe a bit lazy of me